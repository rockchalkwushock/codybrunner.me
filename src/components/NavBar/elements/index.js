import Nav from './Nav'
import NavIcon from './NavIcon'
import NavList from './NavList'

export { Nav, NavIcon, NavList }
