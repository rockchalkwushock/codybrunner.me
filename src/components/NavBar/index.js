import React, { Component } from 'react'
import { arrayOf, number, shape, string } from 'prop-types'

import { Nav, NavIcon, NavList } from './elements'
import { Icon, Link } from '../commons'

/**
 * @fileOverview
 * Site Navigation
 *
 * Present in all views.
 */

class NavBar extends Component {
  static propTypes = {
    menu: arrayOf(
      shape({
        href: string.isRequired,
        id: number.isRequired,
        text: string.isRequired
      })
    ).isRequired
  }
  state = {
    visible: false
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.location !== this.props.location) {
      // Handle different route navigation & closing menu.
      this.setState(state => ({ ...state, visible: false }))
    }
    // Handles traveling to same route & closing menu.
    this.setState(state => ({ ...state, visible: false }))
  }
  handleOnClick = () =>
    this.state.visible
      ? this.setState(state => ({ ...state, visible: false }))
      : this.setState(state => ({ ...state, visible: true }))
  renderLinks = () =>
    this.props.menu.map(l => <Link href={l.href} key={l.id} text={l.text} />)
  render() {
    return (
      <Nav>
        <NavIcon aria-label="hamburger button" onClick={this.handleOnClick}>
          <Icon icon="bars" size="3x" />
        </NavIcon>
        <NavList visible={this.state.visible}>{this.renderLinks()}</NavList>
      </Nav>
    )
  }
}

export default NavBar
