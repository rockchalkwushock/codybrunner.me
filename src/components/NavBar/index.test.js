import React from 'react'
import { shallow } from 'enzyme'

import NavBar from './'

const { siteMetadata } = require('../../../gatsby-config')

describe('Component: <NavBar />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<NavBar menu={siteMetadata.menu} location="/about" />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should present as not visible for mobile nav', () => {
    expect(wrapper.state()).toHaveProperty('visible', false)
  })
  test('should toggle state onClick', () => {
    expect(wrapper.state()).toHaveProperty('visible', false)
    wrapper.find('NavIcon').simulate('click')
    expect(wrapper.state()).toHaveProperty('visible', true)
    wrapper.find('NavIcon').simulate('click')
    expect(wrapper.state()).toHaveProperty('visible', false)
  })
})
