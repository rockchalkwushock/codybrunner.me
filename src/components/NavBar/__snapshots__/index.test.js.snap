// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <NavBar /> should render correctly 1`] = `
<Nav>
  <NavIcon
    aria-label="hamburger button"
    onClick={[Function]}
  >
    <SiteIcon
      icon="bars"
      size="3x"
    />
  </NavIcon>
  <NavList
    visible={false}
  >
    <SiteLink
      href="/"
      key="1"
      text="Home"
    />
    <SiteLink
      href="/about"
      key="2"
      text="About"
    />
    <SiteLink
      href="/posts/1"
      key="3"
      text="Blog"
    />
    <SiteLink
      href="/certs"
      key="4"
      text="Certifications"
    />
  </NavList>
</Nav>
`;
